### PR TITLE
Add buildSearchableOptions and jarSearchableOptions tasks

### PR DESCRIPTION
**Pull request by artspb**
Let's start a review process.

### CODE REVIEW

File: src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy [hunk 1]
```diff
@@ -312,44 +319,92 @@ class IntelliJPlugin implements Plugin<Project> {
...
-                return Jvm.current().javaExecutable.absolutePath
-            })
+                LOG.warn("Cannot resolve builtin JBRE $builtinJbreVersion. Falling local Java.")
+            }
+            return Jvm.current().javaExecutable.absolutePath
+        })
+    }
 
-            task.dependsOn(PREPARE_SANDBOX_TASK_NAME)
+    private static void configureJarSearchableOptions(@NotNull Project project) {
```
Comment (zolotov) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: please add `Task` suffix to the method name

File: src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy [hunk 2]
```diff
@@ -312,44 +319,92 @@ class IntelliJPlugin implements Plugin<Project> {
...
-                    if (jbre != null) {
-                        return jbre.javaExecutable
-                    }
-                    LOG.warn("Cannot resolve JBRE $jbreVersion. Falling back to builtin JBRE.")
+            prepareConventionMappingsForRunIdeTask(project, extension, task)
+            task.dependsOn(PREPARE_SANDBOX_TASK_NAME)
+        }
+    }
+
+    private static void configureBuildSearchableOptions(@NotNull Project project, @NotNull IntelliJPluginExtension extension) {
```
Comment (zolotov) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: Please add Task suffix to the method name

File: src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy (hunk 1 above)
Comment (zolotov) [inline src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy]: I think it make sense to extract it into a separate class to make `IntellijPlugin` class smaller

**Comment by artspb** [comment]
Done. I'll add more tests when we have a public build of 2019.1.

**Comment by ice1000** [comment]
I'm having issues using the `buildSearchableOptions` task. My OS is Ubuntu 18.04, with OpenJDK 10.0.2.

Exception:
```
> Task :buildSearchableOptions
OpenJDK 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.ide.ClassUtilCore to field sun.net.www.protocol.jar.JarFileFactory.fileCache
WARNING: Please consider reporting this to the maintainers of com.intellij.ide.ClassUtilCore
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Starting searchable options index builder
Searchable options index builder failed
java.lang.NullPointerException
	at com.intellij.openapi.options.ex.ConfigurableWrapper.disposeUIResources(ConfigurableWrapper.java:174)
	at com.intellij.ide.ui.search.TraverseUIStarter.startup(TraverseUIStarter.java:156)
	at com.intellij.ide.ui.search.TraverseUIStarter.main(TraverseUIStarter.java:87)
	at com.intellij.idea.IdeaApplication.lambda$run$1(IdeaApplication.java:211)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.idea.IdeaApplication.run(IdeaApplication.java:211)
	at com.intellij.idea.IdeaApplication.lambda$initApplication$0(IdeaApplication.java:74)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:322)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

> Task :buildSearchableOptions FAILED
```

**Comment by ice1000** [comment]
Local IDE version is PyCharm-C `191.4212.43`

**Comment by artspb** [comment]
@ice1000 I'd appreciate if you file a separate issue and attach IDE's log files there, they should be in `build/idea-sandbox/system/log`.

**Pull request merged**